src/vmm/src/lib.rs

```diff
@@ -1093,4 +1128,98 @@ mod tests {
             assert!(fdt.write_to_mem(&vmm.guest_memory, fdt_offset).is_err());
         }
     }
+    #[test]
+    fn test_address_alloc() {
```

Comment:
Let's re-write this test to avoid duplication. To do that, we need to identify what are the things that are different on different platforms. Once we do that, we need to find a common name for the things that are different, and then just initialize them with separate values depending on the platform. We can also make some simplifications, so that we don't need to define a lot of variables.

For example, let's say we have a `start_addr` from which addresses are valid. Then, we can define an `invalid_addr` as being `start_addr - 100`. While `start_addr` needs to be defined once for each platform, the `invalid_addr` will just be defined once for both platforms.